src/webhooks/types.ts

```diff
@@ -9,15 +9,22 @@ type WebhookHandlerFunction = (
   body: string
 ) => Promise<void>;
 
-export interface RegisterOptions {
+export type RegisterOptions = {
```

Comment:
I really like the way you set up these types, but I'd consider putting the common fields into a private `RegisterBaseOptions` type so we can more easily evolve this in the future.

Reply:
Yes, completely agree! I had the exact same thought about using an extendable base type for options but it was about 2am when I wrote that so I took the lazy route.

Reply:
Alternatively, if we remove `EVENTBRIDGE_ARN` from `Context`, path would no longer be optional when `deliveryMethod` is of type `DeliveryMethod.EventBridge`. So this could be reverted to the previous interface with an extra `deliveryMethod` property of type `DeliveryMethod` anyway.